sandbox/gamedev/unity/samples/azure-facebook-auth-unity.md

```diff
@@ -0,0 +1,646 @@
+
+7. Click **Save Changes**.
+
+  ![Android platform settings](media/fbauth_fb-android-settings.png)
+
+## Try the test scene
+> [!NOTE]
+> The Facebook Unity SDK does not support logging in in the Unity editor. To test in the editor, paste a [debug access token](https://developers.facebook.com/tools/accesstoken/) into the `LogInUser()` function in `FacebookLogin.cs`.
+
+1. Download and import the [Azure Easy Tables Client with Facebook auth Unity package](https://github.com/dantogno/Azure-Unity-FB-Auth/raw/master/Unity%20Packages/Azure%20Easy%20tables%20client%20with%20FB%20auth.unitypackage) into your Unity project.
+
+2. In the [Azure portal](https://portal.azure.com), navigate to the Function App you created for this example and copy the URL.
+
+  ![Copy function app URL](media/fbauth_copy-functionapp-url.png)
+
+3. Open the file **EasyTableClient.cs** and change `private const string url = "REPLACE_WITH_YOUR_FUNCTIONAPP_URL"` to match the URL of the function app you created for this example, with `/api/` appended at the end.
+> [!IMPORTANT]
+> Be sure to append `/api/` to the end of your Function App URL!
+
+4. In the Unity menu, choose **File > Build Settings...** and add the `Assets/Azure Easy tables client with FB Auth/Test Scene/test scene.unity` file to the index 0 position of the **Scenes in Build** list.
+
+5. Ensure a platform supported by the Facebook Unity SDK is selected (Android, iOS, WebGL).
+
+6. For iOS or Android, choose **Build and Run**. Choose an appropriate path to save the build. For WebGL click **Build** and upload the build to a hosting location on the web.
+
+7. Once the game is running, press the **LOGIN** button.
+
+  ![test login](media/fbauth_test-login.png)
+
+8. This should present a Facebook login screen. Complete the Facebook login process.
+
+  ![facebook login](media/fbauth_fb-login.png)
+
+9. Press **INSERT**. This will add test data to the TestPlayerData table on your Azure Mobile App. Observe the **Output** window to see if the insert operation completed.
+
+10. Press the **GET ALL ENTRIES** button. The **Output** window should print the count of entries in the TestPlayerData table. Verify that the count goes up after each new insert. Additionally, you can navigate to the TestPlayerData table in the Easy Tables section of your Mobile App on the [Azure portal](https://portal.azure.com) and manually inspect the entries.
+
+> [!NOTE]
+> The App Service will time out after a period of inactivity. It may take a moment to spin back up when it is hit again, so if the first test fails, please try again to ensure the service is in its running state.
+
+## Try the sample racing game
+
+1. Download and import the [Azure Easy Tables with FB auth sample game assets Unity package](https://github.com/dantogno/Azure-Unity-FB-Auth/raw/master/Unity%20Packages/Azure%20Easy%20tables%20with%20FB%20auth%20sample%20game%20assets.unitypackage) into your Unity project.
+
```

Comment:
Getting an error on EasyTablesObjectBase not found in the unitypackage?  Also, let's figure out the best way to get this over to the AzureSamples-Unity repo (we'll be moving all this stuff to Azure-Samples org at some point in the near future)

Reply:
Though I see in the readme.txt of the package there's a 2nd package to download/import?  Did I skip over that here?

Reply:
I did.  :)  Are these all in a single repo?  Or can we make this easier for folks running through the demo?  I like how we did the first one where everything is in a single repo the user can clone.

Reply:
And perhaps we should think about how the previous article links to this one and vice-versa?  Is this article perhaps just "Part 2" where we add on to what was built in the previous?

Reply:
These are tough questions I sort of consciously pushed to the backburner in order to just move forward at the time...

It would be great to set the two docs up as part 1 and part 2. I think that would help break this up into a more digestible size.  Unfortunately, however, they use different approaches so it's difficult to decide how easily they fit together. 

The first doc uses the Azure Mobile Apps SDK for Unity to read and write from the databases. This is definitely the most straightforward approach, but there are some issues that caused me to change the approach for the facebook example. **Most notably, the Facebook SDK for Unity doesn't work with the .NET 4.6 equivalent scripting runtime.**  So this is why I am offloading calls to the mobile apps sdk to Azure Functions. This is more complicated and screws up the part 1, part 2 relationship of the two docs, but it achieves compatibility with every platform.

I guess one day FB will have to update to support the new scripting runtime, but I'm not sure what kind of timeline that would be. I guess my thoughts are that this document can be useful in the meantime, and then perhaps I can revisit the "part 1, part 2" approach in the future once everything is using the new scripting runtime. Let me know if this makes sense or if you have any other ideas :)